### DISCUSSION TITLE
VueUse window is not defined

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#17903] [Questions] [answered]
Any can help? why does it work intermittently
I have some element to which I am trying to set dimensions and bind to the resize event:
`@/components/Element.vue`
```js
import { Ref } from 'vue';
import { useEventListener } from '@vueuse/core';

const ElementWrapper: Ref<HTMLElement> = ref(null);

function setElementHeight() {
    ElementWrapper.value.style.height = (window.innerWidth * 39) / 100 + 'px';
}

useEventListener(window, 'resize', setElementHeight);
```

And periodically I get an error in the console: 
- `[Vue warn]: Unhandled error during execution of setup function at <Element> window is not defined`
- ` ERROR  [worker] window is not defined  `

**Comment by Aareksio** [comment, marked as answer]
https://github.com/vueuse/vueuse/blob/main/packages/core/useEventListener/index.ts#L90
https://github.com/vueuse/vueuse/blob/main/packages/core/useEventListener/index.ts#L101_L116

I am positive you could either do `useEventListener('resize', setElementHeight)` or import `_defaultWindow` yourself and use it instead of `window`.
